code/modules/projectiles/updated_projectiles/gun_attachables.dm

```diff
@@ -83,6 +83,10 @@ Defined in conflicts.dm of the #defines folder.
 	var/attachment_action_type
 	var/scope_zoom_mod = FALSE //codex
 
+	var/ammo_mod = null			//what ammo the gun could fire if overcharge is activated, lasers usually.
```

Comment:
why are you doing this in the attachment

Reply:
Same Variable name, but they were both local. Changed the gun one to ammo_diff.